Build is not kicking off in a browser on save

### OS
Windows 10.

### Versions.
```
angular-cli: 1.0.0-beta.28.3
node: 6.9.2
os: win32 x64
@angular/common: 2.4.5
@angular/compiler: 2.4.5
@angular/core: 2.4.5
@angular/flex-layout: 2.0.0-beta.4
@angular/forms: 2.4.5
@angular/http: 2.4.5
@angular/material: 2.0.0-beta.1
@angular/platform-browser: 2.4.5
@angular/platform-browser-dynamic: 2.4.5
@angular/router: 3.4.5
@angular/compiler-cli: 2.4.5
```


### Repro steps.

Migrate from beta.24 to beta.28.3
Modify any file
Click save
Chrome dev console saying:
```
[WDS] App updated. Recompiling...
[WDS] Nothing changed.
```
The new build is not kicking off

So had to stop and reastart `ng serve` manually

I freshly installed angular-cli `npm uninstall -g angular-cli && npm i -g angular-cli` and im getting the following warnings:
```
npm WARN optional Skipping failed optional dependency /angular-cli/chokidar/fsevents:
npm WARN notsup Not compatible with your operating system or architecture: fsevents@1.0.17
npm WARN @angular/compiler-cli@0.6.4 requires a peer of @angular/compiler@2.0.2 but none was installed.
npm WARN @angular/compiler-cli@0.6.4 requires a peer of @angular/platform-server@2.0.2 but none was installed.
npm WARN @angular/compiler-cli@0.6.4 requires a peer of @angular/core@2.0.2 but none was installed.
npm WARN @angular/core@2.4.5 requires a peer of rxjs@^5.0.1 but none was installed.
npm WARN @angular/core@2.4.5 requires a peer of zone.js@^0.7.2 but none was installed.
```

Trying `ng --version` gives me a lot of `Cannot find module 'reflect-metadata'` Errors.
Is this related to the restructuring of the cli packages?

Previously it worked just fine.

@web-dave: tried your fix and now get the following on calling ng serve:

Cannot find module 'rxjs/symbol/observable'
Error: Cannot find module 'rxjs/symbol/observable'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at globalScope (C:\Users\Media\AppData\Roaming\npm\node_modules\@angular\cli\node_modules\@angular\core\bundles\core.umd.js:7:85)
    at Object.<anonymous> (C:\Users\Media\AppData\Roaming\npm\node_modules\@angular\cli\node_modules\@angular\core\bundles\core.umd.js:10:2)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)

Output from ng v:

@angular/cli: 1.0.0-beta.30
node: 6.9.4
os: win32 x64
@angular/common: 2.4.6
@angular/compiler: 2.4.6
@angular/core: 2.4.6
@angular/forms: 2.4.6
@angular/http: 2.4.6
@angular/platform-browser: 2.4.6
@angular/platform-browser-dynamic: 2.4.6
@angular/router: 3.4.6
@angular/compiler-cli: 2.4.6
@ngtools/webpack: 1.2.4

--- a/packages/@ngtools/webpack/src/compiler_host.ts
+++ b/packages/@ngtools/webpack/src/compiler_host.ts
@@ -187,6 +187,7 @@ export class WebpackCompilerHost implements ts.CompilerHost {
   }
 
   invalidate(fileName: string): void {
+    fileName = this._resolve(fileName);
     if (fileName in this._files) {
       this._files[fileName] = null;
       this._changedFiles[fileName] = true;

